.github/workflows/build-image-job.yaml

```diff
@@ -0,0 +1,128 @@
+name: Build docker image
+permissions:
+  actions: read
+  checks: read
+  contents: write
+  deployments: read
+  discussions: read
+  id-token: write
+  issues: read
+  packages: read
+  pages: read
+  pull-requests: read
+  repository-projects: read
+  statuses: read
+  security-events: read
+
+on:
+  workflow_call:
+    inputs:
+      dockerfile:
+        description: 'Path to the Dockerfile'
+        required: true
+        type: string
+      registry:
+        description: 'Docker registry to push to'
+        required: false
+        default: 'docker.io'
+        type: string
+      image:
+        description: 'Name of the image'
+        required: true
+        type: string
+      image_tag:
+        description: 'Tag of the image'
+        required: true
+        type: string
+      push:
+        description: 'Push the image to the registry'
+        required: false
+        type: boolean
+        default: true
+      platforms:
+        description: 'Platforms to build for'
+        required: false
+        default: 'linux/amd64'
+        type: string
+      runs-on:
+        description: 'The type of machine to run the job on'
+        required: false
+        default: 'ubuntu-latest'
+        type: string
+    secrets:
+      username: 
+        description: 'Docker registry username'
+        required: false
+      password:
+        description: 'Docker registry token (password)'
+        required: false
+
+jobs:
+  docker:
+    runs-on: ${{ inputs.runs-on }}
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Print inputs
+        run: |
+          echo "dockerfile: ${{ inputs.dockerfile }}"
+          echo "registry: ${{ inputs.registry }}"
+          echo "image: ${{ inputs.image }}"
+          echo "image_tag: ${{ inputs.image_tag }}"
+          echo "push: ${{ inputs.push }}"
+          echo "platforms: ${{ inputs.platforms }}"
+          
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      
+      # docker is not installed on MacOS - https://github.com/actions/runner-images/issues/2150 
+      - name: Setup docker (missing on MacOS)
+        if: runner.os == 'macos'
+        run: |
+          brew install docker colima
+          colima start
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
```

Comment:
Do we still need buildx?

Reply:
or do we want to keep this one to do both platforms if needed in the future?

Reply:
We still use it for arm64 builds